src/main/java/firebase/sdk/dashboard/dao/SDKDaoDatastore.java

```diff
@@ -0,0 +1,321 @@
+    List<VersionMetadata> sdkVersionHistory = getVersionHistoryFromProperty(
+        (List<EmbeddedEntity>) entity.getProperty("versionHistory"));
+
+    SDK sdk = SDK.newBuilder()
+      .platform(sdkPlatform)
+      .libraryName(sdkLibraryName)
+      .libraryGroup(sdkLibraryGroup)
+      .externalName(sdkExternalName)
+      .fireEscapeName(sdkFireEscapeName)
+      .owner(sdkOwner)
+      .versionHistory(sdkVersionHistory)
+      .build();
+
+    return sdk;
+  }
+
+  public List<String> getSDKs(Platform platform) {
+    FilterPredicate platformPropertyFilter = makePropertyFilter("platform", platform.getLabel());
+    Query query = new Query("SDK")
+      .setFilter(platformPropertyFilter);
+
+    PreparedQuery preparedQuery = DATASTORE.prepare(query);
+    FetchOptions fetchOptions = FetchOptions.Builder.withDefaults();
+    QueryResultIterable<Entity> results = preparedQuery.asQueryResultIterable();
+    if (results == null) {
+      return Arrays.asList();
+    }
+
+    ArrayList<String> sdksForPlatform = new ArrayList<>();
+    for (Entity entity : results) {
+      sdksForPlatform.add(
+          (String) entity.getProperty("libraryName"));
+    }
+    return sdksForPlatform;
+  }
+
+  public List<String> getSDKsEnrolledInRelease(Platform platform, String releaseName) {
+    FilterPredicate platformPropertyFilter = makePropertyFilter("platform", platform.getLabel());
+    FilterPredicate releasePropertyFilter = makePropertyFilter("releaseName", releaseName);
+    CompositeFilter compositeFilter = new CompositeFilter(
+        CompositeFilterOperator.AND, Arrays.asList(
+          platformPropertyFilter,
+          releasePropertyFilter));
+
+    Query query = new Query("SDKReleaseMetadata")
+      .setFilter(compositeFilter);
+
+    PreparedQuery preparedQuery = DATASTORE.prepare(query);
+    FetchOptions fetchOptions = FetchOptions.Builder.withDefaults();
+    QueryResultIterable<Entity> results = preparedQuery.asQueryResultIterable(fetchOptions);
+    if (results == null) {
+      return Arrays.asList();
+    }
+
+    ArrayList<String> enrolledSDKs = new ArrayList<>();
+    for (Entity entity : results) {
+      String libraryName = (String) entity.getProperty("libraryName");
+      enrolledSDKs.add(libraryName);
+    }
+    return enrolledSDKs;
+  }
+
+  public SDKReleaseMetadata getSDKReleaseMetadata(Platform platform, String releaseName, String sdkName) {
+    Key sdkReleaseMetadataKey = KeyFactory.createKey("SDKReleaseMetadata", platform.getLabel() + "_" + sdkName + "_" + releaseName);
+    FilterPredicate keyPropertyFilter = makePropertyFilter("__key__", sdkReleaseMetadataKey);
+
+    Query query = new Query("SDKReleaseMetadata")
+      .setFilter(keyPropertyFilter);
+
+    PreparedQuery preparedQuery = DATASTORE.prepare(query);
+    Entity entity = preparedQuery.asSingleEntity();
+    if (entity == null) {
+      return null;
+    }
+
+    Platform srmPlatform = Platform.get((String) entity.getProperty("platform"));
```

Comment:
readable variable names

Reply:
Ah, I see why you used givenPlatform before :). You could actually avoid defining these intermediate variables and add code to the builder. Or use sdkPlatform, etc like in the getSDK method.